src/jit/assertionprop.cpp

```diff
@@ -3912,6 +3913,26 @@ GenTree* Compiler::optAssertionProp_Update(GenTree* newTree, GenTree* tree, GenT
             // optAssertionPropMain(). It will reset the gtPrev and gtNext links for all nodes.
             newTree->gtNext = tree->gtNext;
 
+            if ((linkData.parent != nullptr) && linkData.parent->IsCall())
+            {
+                GenTreeCall* parentCall = linkData.parent->AsCall();
```

Comment:
I would prefer to see this leverage the code in `GenTreeCall::ReplaceCallOperand`. Although it is primarily used in `LIR`, I don't think it depends on it, and it is a more thorough handling of calls. I'm not crazy about the use of `TryGetUse`: https://github.com/dotnet/coreclr/blob/master/src/jit/gentree.cpp#L1041
```
    assert(TryGetUse(*useEdge, &useEdge));
```
which overwrites its argument without verifying that it gets the same result (and `TryGetUse` is primarily an `LIR` thing, but basically just searches for a parent edge). But I don't think we should add this duplicate functionality.